File: buerguer-pig/HamburgesasEIngredientes-Java-IvanRC/src/main/java/storage/hamburgesa/HamburgesaStorageServiceSerializable.java
package storage.hamburgesa;

import config.ConfigApp;
import models.Hamburgesa;

import java.io.*;
import java.util.ArrayList;
import java.util.List;

public class HamburgesaStorageServiceSerializable implements HamburgesaStorageService {

    ConfigApp configApp = ConfigApp.getInstance();

    File file = new File(configApp.getAPP_DATA()+File.separator+"hamburgesa.ser");

    public HamburgesaStorageServiceSerializable() throws IOException {}

    @Override
    public void saveAll(List<Hamburgesa> entities) throws IOException {
        try(ObjectOutputStream objectOutputStream = new ObjectOutputStream(new FileOutputStream(file))){
            objectOutputStream.writeObject(entities);
        }
    }

    @Override
    public List<Hamburgesa> loadAll() throws IOException {
        List<Hamburgesa> hamburgesas = new ArrayList<>();
        if(!file.exists()) return hamburgesas;
        try(ObjectInputStream objectInputStream = new ObjectInputStream(new FileInputStream(file))){
            hamburgesas = (List<Hamburgesa>) objectInputStream.readObject();
        } catch (ClassNotFoundException e) {
            throw new RuntimeException(e);
        }
        return hamburgesas;
    }
}

File: buerguer-pig/HamburgesasEIngredientes-Kotlin-IvanRC/src/main/kotlin/storageService/Hamburger/HamburgesaStorageServiceSerializable.kt
package storageService.Hamburger

import config.ConfigApp
import model.Hamburgesa
import model.Ingrediente
import mu.KotlinLogging
import java.io.*

class HamburgesaStorageServiceSerializable: HamburgesaStorageService {

    private val config = ConfigApp
    private val file = File(config.APP_DATA+ File.separator+"hamburgesa.ser")
    private val logger = KotlinLogging.logger {}

    override fun saveAll(entities: List<Hamburgesa>) {
        logger.debug { "Se guardan todos las hamburgesas en el fichero serializable." }
        ObjectOutputStream(FileOutputStream(file)).use{
            it.writeObject(entities)
        }
    }

    override fun loadAll(): List<Hamburgesa> {
        logger.debug { "Se cargán todos las hamburgesas del fichero serializable." }
        var hamburgesas = mutableListOf<Hamburgesa>()
        if(!file.exists()) return hamburgesas.toList()
        ObjectInputStream(FileInputStream(file)).use{
            hamburgesas = it.readObject() as MutableList<Hamburgesa>
        }
        return hamburgesas.toList()
    }
}